Mage/Task/AbstractTask.php

```diff
@@ -194,9 +194,12 @@ protected function getParameters()
         }
 
         // if general.yml includes "ssy_needs_tty: true", then add "-t" to the ssh command
-        $needs_tty = ($this->getConfig()->general('ssh_needs_tty', false) ? '-t' : '');
+        $ttyIsNeeded = $this->getConfig()->general('ssh_needs_tty', false)
```

Comment:
tty OR ssh agent is required .. so why enforce tty? I could want to have an ssh agent running and avoid tty for some reason ..

My opinion is to enforce as few as possible .. always .. the user can adjust the config as per his needs and environment, without being forced to follow an hardcoded behaviour